sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4319,6 +4319,146 @@ object SQLConf {
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val H2_DIALECT_VERSION = buildConf("spark.sql.connector.h2.dialectVersion")
+    .internal()
+    .doc("Dialect version used for h2 connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val MSSQLSERVER_JDBC_VERSION = buildConf("spark.sql.connector.mssqlserver.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for mssqlserver connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val MSSQLSERVER_DIALECT_VERSION = buildConf("spark.sql.connector.mssqlserver.dialectVersion")
+    .internal()
+    .doc("Dialect version used for mssqlserver connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val MYSQL_JDBC_VERSION = buildConf("spark.sql.connector.mysql.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for mysql connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val MYSQL_DIALECT_VERSION = buildConf("spark.sql.connector.mysql.dialectVersion")
+    .internal()
+    .doc("Dialect version used for mysql connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val ORACLE_JDBC_VERSION = buildConf("spark.sql.connector.oracle.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for oracle connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val ORACLE_DIALECT_VERSION = buildConf("spark.sql.Dialect oracle.dialectVersion")
+    .internal()
+    .doc("JDBC version used for oracle connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val POSTGRES_JDBC_VERSION = buildConf("spark.sql.connector.postgres.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for postgres connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val POSTGRES_DIALECT_VERSION = buildConf("spark.sql.connector.postgres dialectVersion")
+    .internal()
+    .doc("JDBC version used for postgres connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val SNOWFLAKE_JDBC_VERSION = buildConf("spark.sql.connector.snowflake.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for snowflake connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val SNOWFLAKE_DIALECT_VERSION = buildConf("spark.sql.connector.snowflake.dialectVersion")
+    .internal()
+    .doc("Dialect version used for snowflake connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val TERADATA_JDBC_VERSION = buildConf("spark.sql.connector.teradata.jdbcVersion")
+    .internal()
+    .doc("JDBC version used for teradata connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
+
+  val TERADATA_DIALECT_VERSION = buildConf("spark.sql.connector.teradata.dialectVersion")
+    .internal()
+    .doc("Dialect version used for teradata connector.")
+    .version("4.0.0")
+    .intConf
+    .createWithDefault(0)
```

Comment:
Do we need major and minor versions here, maybe string value is better for versioning